core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -307,6 +377,56 @@ public String encryptAndBase64Encode(byte[] plainText) throws
         }
     }
 
+    /**
+     * Decrypt the cipher text using the given algorithm and internal crypto provider type.
+     *
+     * @param cipherTextBytes            The cipher text to be decrypted.
+     * @param algorithm                  The algorithm used for decryption.
+     * @param internalCryptoProviderType The {@link org.wso2.carbon.crypto.api.InternalCryptoProvider} type.
+     * @return Decrypted bytes.
+     * @throws CryptoException
+     */
+    public byte[] decrypt(byte[] cipherTextBytes, String algorithm, String internalCryptoProviderType)
+            throws CryptoException {
+
+        failIfEncryptDecryptInputsAreInvalid(cipherTextBytes,algorithm,internalCryptoProviderType);
+
+        byte[] decryptedValue;
+
+        try {
+            CryptoService cryptoService = CarbonCoreDataHolder.getInstance().getCryptoService();
+
+            if (cryptoService == null) {
+                throw new CryptoException("A crypto service implementation has not been registered.");
+            }
+
+            CipherMetaDataHolder
+                    cipherMetaDataHolder = cipherTextToCipherMetaDataHolder(cipherTextBytes);
+            if (cipherMetaDataHolder != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            String.format("Cipher text is in self contained format. Retrieve the actual cipher from " +
+                                    "the self contained cipher text."));
+                }
+                cipherTextBytes = cipherMetaDataHolder.getCipherBase64Decoded();
+            }
+
+            if (cipherTextBytes.length == 0) {
+                decryptedValue = StringUtils.EMPTY.getBytes();
+                if (log.isDebugEnabled()) {
+                    log.debug("Ciphertext is empty. An empty array will be used as the plaintext bytes.");
+                }
+            } else {
+                decryptedValue = cryptoService.decrypt(cipherTextBytes, algorithm, CRYPTO_API_PROVIDER_BC,
+                        internalCryptoProviderType);
+            }
+            return decryptedValue;
+        } catch (Exception e) {
```

Comment:
Can't we catch the specific exception here?

Reply:
Fixed with 7157b37d5c65be094165fc2ee48137ca915b81c3